src/core/libraries/kernel/thread_management.cpp

```diff
@@ -1373,97 +1375,90 @@ int PS4_SYSV_ABI posix_pthread_detach(ScePthread thread) {
     return pthread_detach(thread->pth);
 }
 
-int PS4_SYSV_ABI posix_sem_init(PthreadSemInternal** sem, int pshared, unsigned int value) {
-    if (value > ORBIS_KERNEL_SEM_VALUE_MAX) {
-        SetPosixErrno(EINVAL);
-        return -1;
-    }
-    if (sem != nullptr) {
-        *sem = new PthreadSemInternal{
-            .semaphore = std::counting_semaphore<ORBIS_KERNEL_SEM_VALUE_MAX>{value},
-            .value = {static_cast<int>(value)},
-        };
+int PS4_SYSV_ABI posix_sem_init(sem_t* sem, int pshared, unsigned int value) {
+    int result = sem_init(sem, pshared, value);
```

Comment:
This cannot be used on macOS, unnamed POSIX semaphores are not implemented.

Reply:
Oh, this was just an experiment to make `evStart` from `pthreads` to alert the thread. This should be removed and re-worked either way. We cannot use `winpthreads` anyway as it does not provide the required functionality.